WalletWasabi/Wallets/Wallet.cs

```diff
@@ -118,14 +119,14 @@ private set
 
 	public Task<IEnumerable<SmartTransaction>> GetTransactionsAsync() => Task.FromResult(GetTransactions());
 
-	public Task<IEnumerable<SmartCoin>> GetCoinjoinCoinCandidatesAsync() => Task.FromResult(GetCoinjoinCoinCandidates());
+	public Task<CoinsRegistry> GetCoinjoinCoinCandidatesAsync() => Task.FromResult(GetCoinjoinCoinCandidates());
 
-	public IEnumerable<SmartCoin> GetCoinjoinCoinCandidates() => Coins;
+	public CoinsRegistry GetCoinjoinCoinCandidates() => Coins;
```

Comment:
Why don't we check the banned coins here in the `Wallet`?

Reply:
That can be done. 
Standing currently, it can be
```cs
public IEnumerable<SmartCoin> GetCoinjoinCoinCandidates() => Coins.Where(coin => !Coins.TryGetBannedCoin(coin, DateTimeOffset.UtcNow, out _));
```
so in the CoinJoinManager, we are not dependant on CoinsRegistry. That way we can remove the questionable interface change that [could hurt affiliation integrations](https://github.com/zkSNACKs/WalletWasabi/pull/10825/files#r1229789584).

Only downside I can think is that it won't be consistent in a meaning that not all checks for coinjoin-availability will be done at the same place(CoinJoinManager), which could lead to confusion in the future. @lontivero what do you think?